logstash-core/lib/logstash/filter_delegator.rb

```diff
@@ -39,9 +39,11 @@ def config_name
     def multi_filter(events)
       @metric_events.increment(:in, events.size)
 
-      clock = @metric_events.time(:duration_in_millis)
+      start_time = java.lang.System.current_time_millis
       new_events = @filter.multi_filter(events)
-      clock.stop
+      @metric_events.report_time(
+        :duration_in_millis, java.lang.System.current_time_millis - start_time
```

Comment:
(This would be a great case to use macros, but Java and Ruby don't have them, sadly)